src/audio/host.c

```diff
@@ -550,11 +557,11 @@ static int host_params(struct comp_dev *dev)
 	 * note: stream_tag is ignored by dw-dma
 	 */
 	hd->chan = dma_channel_get(hd->dma, 0);
+#endif
```

Comment:
Is moving the endif here necessary? The code looks like this now

````c
#if defined CONFIG_DMA_GW
	hd->chan = dma_channel_get(hd->dma, dev->params.stream_tag);
	if (hd->chan < 0) {
		trace_host_error("host_params() error: hd->chan < 0");
		return -ENODEV;
	}
	err = dma_set_config(hd->dma, hd->chan, &hd->config);
	if (err < 0) {
		trace_host_error("host_params() error: "
				 "dma_set_config() failed");
		dma_channel_put(hd->dma, hd->chan);
		return err;
	}
#else
	host_elements_reset(dev);
	/* get DMA channel from DMAC
	 * note: stream_tag is ignored by dw-dma
	 */
	hd->chan = dma_channel_get(hd->dma, 0);
#endif
	if (hd->chan < 0) {
		trace_host_error("host_params() error: hd->chan < 0");
		return -ENODEV;
	}
````

The last test isn't really necessary in the CONFIG_DMA_GW case, is it? hd->chan isn't modified since the first tests, and all errors cases return

Reply:
True , i had an idea here for a slight refactor , but i missed it and left in for final push.